usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -201,6 +218,19 @@ zap_name_alloc(zap_t *zap, const char *key, matchtype_t mt)
 	}
 
 	zn->zn_hash = zap_hash(zn);
+
+	if (zap->zap_normflags) {
+		/*
+		 * We *must* use zn_normflags because this normalization is
+		 * what the matching is based on.  (Not the hash!)
+		 */
+		if (zap_normalize(zap, key, zn->zn_normbuf,
```

Comment:
I don't think that it's an issue, the only possible difference between zap->zap_normflags and zn->zn_normflags is U8_TEXTPREP_TOUPPER flag, which shouldn't affect the resulting length.